File: Edu-Java/src/com/jetbrains/edu/java/JCourseProjectGenerator.java
package com.jetbrains.edu.java;

import com.intellij.openapi.module.ModifiableModuleModel;
import com.intellij.openapi.module.Module;
import com.intellij.openapi.module.ModuleWithNameAlreadyExists;
import com.intellij.openapi.options.ConfigurationException;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.util.InvalidDataException;
import com.jetbrains.edu.learning.courseFormat.Course;
import com.jetbrains.edu.learning.intellij.generation.CourseModuleBuilder;
import com.jetbrains.edu.learning.intellij.generation.IntellijCourseProjectGeneratorBase;
import com.jetbrains.edu.learning.intellij.generation.EduModuleBuilderUtils;
import org.jdom.JDOMException;
import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public class JCourseProjectGenerator extends IntellijCourseProjectGeneratorBase {

  public JCourseProjectGenerator(@NotNull Course course) {
    super(course);
  }

  @NotNull
  @Override
  protected CourseModuleBuilder studyModuleBuilder() {
    return new CourseModuleBuilder() {
      @NotNull
      @Override
      public Module createModule(@NotNull ModifiableModuleModel moduleModel) throws InvalidDataException, IOException, ModuleWithNameAlreadyExists, JDOMException, ConfigurationException {
        Module baseModule = super.createModule(moduleModel);
        Project project = baseModule.getProject();
        EduModuleBuilderUtils.createCourseFromCourseInfo(moduleModel, project, myCourse, getModuleFileDirectory());
        return baseModule;
      }
    };
  }
}

File: Edu-Java/src/com/jetbrains/edu/java/JCourseProjectGenerator.kt
package com.jetbrains.edu.java

import com.intellij.lang.java.JavaLanguage
import com.intellij.openapi.project.Project
import com.intellij.psi.codeStyle.CodeStyleSettingsManager
import com.jetbrains.edu.learning.courseFormat.Course
import com.jetbrains.edu.learning.intellij.GradleCourseBuilderBase
import com.jetbrains.edu.learning.intellij.JdkProjectSettings
import com.jetbrains.edu.learning.intellij.generation.GradleCourseProjectGenerator

class JCourseProjectGenerator(builder: GradleCourseBuilderBase,course: Course) : GradleCourseProjectGenerator(builder, course) {

  override fun afterProjectGenerated(project: Project, projectSettings: JdkProjectSettings) {
    super.afterProjectGenerated(project, projectSettings)
    val indentOptions = CodeStyleSettingsManager.getInstance().currentSettings.
                                                                  getCommonSettings(JavaLanguage.INSTANCE).indentOptions
    indentOptions?.TAB_SIZE = 2
    indentOptions?.INDENT_SIZE = 2
  }
}
